src/schema/city/index.ts

```diff
@@ -53,31 +54,57 @@ const CityType = new GraphQLObjectType<any, ResolverContext>({
         },
       }),
       resolve: async (city, args, { showsWithHeadersLoader }) => {
-        // default Enum value for status is not properly resolved
-        // so we have to manually resolve it by lowercasing the value
-        // https://github.com/apollographql/graphql-tools/issues/715
-        args.status = args.status.toLowerCase()
-
-        const gravityOptions = {
-          ...convertConnectionArgsToGravityArgs(args),
-          displayable: true,
+        const baseParams = {
           near: `${city.coordinates.lat},${city.coordinates.lng}`,
           max_distance: LOCAL_DISCOVERY_RADIUS_KM,
           has_location: true,
-          total_count: true,
+          sort: args.sort,
+          // default Enum value for status is not properly resolved
+          // so we have to manually resolve it by lowercasing the value
+          // https://github.com/apollographql/graphql-tools/issues/715
+          status: args.status.toLowerCase(),
+          // This is to ensure the key never makes its way into the `baseParams`
+          // object if not needed,
+          ...(typeof args.discoverable === "undefined"
+            ? undefined
+            : { discoverable: args.discoverable }),
+          ...(typeof args.discoverable === "undefined"
+            ? { displayable: true }
+            : undefined),
         }
-        delete gravityOptions.page
 
-        if (args.discoverable) {
-          delete gravityOptions.displayable
+        let response
+        let offset
+
+        if (args.first === MAX_GRAPHQL_INT) {
+          // TODO: We could throw an error if the `after` arg is passed, but not
+          //       doing so, for now.
+          offset = 0
+          response = await allViaLoader(showsWithHeadersLoader, {
```

Comment:
I dont have much (any 😒 ) context , but taking a look at the `/shows` endpoint- can potentially return _all_ shows sort of thing (depending on the query params used). So I'm assuming this is being used with params that will generally return a much more reasonable-sized list (all 'discoverable' shows near a location), but just wanting to point it out.

The few places of loading all data via a paginated API are anti-patterns, basically. Not saying to change your plan for this PR (or the project generally). One possible alternative might be to load in an export of the previous night show's data generated via the Fulcrum tasks.

Reply:
Yup, all valid points 👍 We’re aware of the issues and are thinking it through.

PS Is also didn’t add a `all` arg to the field to not encourage it for other uses.